Extensions/Spine/JsExtension.js

```diff
@@ -0,0 +1,343 @@
+        this._pixiObject = new PIXI.Container();
+        this._pixiObject.addChild(this._rect);
+        this._pixiContainer.addChild(this._pixiObject);
+
+        this.loadSpine();
+      }
+
+      static getThumbnail(project, resourcesLoader, objectConfiguration) {
+        return 'CppPlatform/Extensions/spriteicon.png';
+      }
+
+      loadSpine() {
+        const jsonResourceName = this.properties.get('jsonResourceName').getValue();
+        const imageResourceName = this.properties.get('imageResourceName').getValue();
+        const atlasResourceName = this.properties.get('atlasResourceName').getValue();
+
+        this._pixiResourcesLoader
+          .getSpineData(this._project, jsonResourceName, imageResourceName, atlasResourceName)
+          .then((spineData) => {
+            this._spine = new PIXI.Spine(spineData);
+            this._pixiObject.addChild(this._spine);
+            this.update();
+          });
+      }
+
+      update() {
+        this._pixiObject.position.set(this._instance.getX(), this._instance.getY());
+
+        this.setAnimation(this._instance.getRawDoubleProperty('animation'));
+
+        const width = this.getWidth();
+        const height = this.getHeight();
+
+        this._rect.clear();
+        this._rect.beginFill(0xFFFFFF);
+        this._rect.lineStyle(0, 0xFFFFFF);
+        this._rect.drawRect(0, 0, width, height);
+
+        if (this._spine) {
+          const s = this._spine;
+          s.width = width;
+          s.height = height;
+          s.alpha = this.properties.get('opacity').getValue() / 255;
+          const localBounds = s.getLocalBounds(undefined, true);
+          s.position.set(-localBounds.x * s.scale.x, -localBounds.y * s.scale.y);
+        }
+
+        this._pixiObject.calculateBounds();
+      }
+
+      setAnimation(index) {
+        const s = this._spine;
+        const {configuration} = this;
+
+        if (!s || configuration.hasNoAnimations() || index === this._animationIndex) {
+          return;
+        }
+
+        if (!Number.isInteger(index) || index < 0) {
+          index = 0;
+        } else if (configuration.getAnimationsCount() <= index) {
+          index = configuration.getAnimationsCount() - 1;
+        }
+
+        this._animationIndex = index;
+        const animation = configuration.getAnimation(index);
+        const name = animation.getName();
+        const source = animation.getSource();
+        const shouldLoop = animation.shouldLoop();
+
+        // reset scale to track new animation range
+        // if custome size is set it will be reinitialized in update method
+        s.scale.set(1, 1);
+        s.state.setAnimation(0, source, shouldLoop);
+        s.state.tracks[0].trackTime = 0;
+        s.update(0);
+        s.autoUpdate = false;
+        this._initialWidth = s.width;
+        this._initialHeight = s.height;
+      }
+
+      getDefaultWidth() {
+        const scale = +this.properties.get('scale').getValue() || 1;
```

Comment:
You can use Number constructor, parseInt/parseFloat functions instead of unary plus operator